content/spin/javascript-components.md

```diff
@@ -97,6 +97,55 @@ Once a Spin compatible module is created, it can be run using
 $ spin up
 ```
 
+---
+
+## A Quick Note About NPM Scripts
+
+Please note that using pre-built NPM scripts can have different effects on different Operating Systems (OSs). Let's take the `npm run build` command (like [the one in the spin-js-sdk](https://github.com/fermyon/spin-js-sdk/blob/main/examples/javascript/hello_world/package.json)) as an example:
+
+<!-- @nocpy -->
+
+```bash
+"scripts": {
+    "build": "npx webpack --mode=production && mkdir -p target && spin js2wasm -o target/spin-http-js.wasm dist/spin.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  }
+```
+
+The `npm run build` command will work on Linux and macOS. However, on Windows it will create both a `-p` directory and a `target` directory.
+
+On Linux/Unix systems, the `-p` option in the `mkdir` command is designed to prevent an error from occuring in the event that the `target` directory already exists. However, on Windows systems, npm (by default) uses cmd.exe which does not recognize the `-p` option, regarding its `mkdir` command.
+
+If you run `npm run build` on Windows (more than once) the following error will be encountered.
+
+<!-- @nocpy -->
+
+```bash
+A subdirectory or file -p already exists
```

Comment:
@karthik2804 I am sure you have explained this to me _at least_ once already, but could you remind me:

* Why we need `-p` here
* Why we need `mkdir` here instead of `js2wasm` creating any required directories

I seem to recall the answer was some awful webpack detail but it has all slipped my mind.  (I am not sure if we should document it somewhere... it does seem like a question these docs are bound to raise.)

Reply:
- The `-p` option on Unix systems allows the command to run without an error if the directory already exists.
- `js2wasm` can potentially create the directories required which actually sounds like a good plan for the next release.